### DISCUSSION TITLE
JavascriptResponse.Result = null

### DISCUSSION THREAD

**Discussion by reinholdbehringer** [cefsharp/CefSharp#4215] [Q&A]
Hi, 

this is an excellent and very powerful tool! But I ran into some trouble:

I am trying to replace the olde C# webbrowser control with this CefSharp webbrowser, but I ran into difficulties using Javascript functions which are already in an HTML page that I am loading with this browser.
The previous C# webbrower control could just use the HtmlDocument.InvokeScript(functionName, parameters), where functionName is the name of the Javascript function. Areturn value was then returned as s string and could be used further in C#.
 
But with this Cef I have trouble achieving the same.
I am using in C#:
JavascriptResponse response = await chromeBrowser.EvaluateScriptAsync(functionName); 

The response.Success is true, but response.Result is always null or empty. I checked in the JavaScript, and there is a valid return value, but it seems not to reach C#.

What do I need to do to get the correct return value of my JavaScript function into C# when calling this function from within my C# code?

Cheers,
Reinhold

**Comment by reinholdbehringer** [comment]
It seems that this problem has solved itself... I have no idea what is different., but this morning when I tried the identical code, it worked OK, and the response.Result had the desired return value from the JavaScript function. 
I will monitor this, and if the error occurs again, I will post more info about it.
Otherwise, I will close this question, if the result remains stable.

**Reply by amaitland** [reply]
If you have further problems then check the log file, make sure your code works in devtools etc. I'd also suggest testing with some basic examples first, make sure `1 + 1` returns `2` etc.

- https://github.com/cefsharp/CefSharp/wiki/Trouble-Shooting#log-file
- https://github.com/cefsharp/CefSharp/wiki/Trouble-Shooting#javascript-debugging

**Reply by reinholdbehringer** [reply]
Thank you, Alex! So far it works great! I do not know what the original problem was... but I keep monitoring it. Might have to do with internet connection conditions....
